### ISSUE TITLE
On app launch entitlement identifier is lowercased string and not as defined in the web portal

### ISSUE THREAD

**Issue by cherpake** [RevenueCat/purchases-ios#1650]
When launching the app - entitlement identifier appears to be lowered case string "pro" and not as it's defined in the web portal "Pro". Then after fetching current customer info from the web - it does appear as expected "Pro".

The problem is that customer info (including entitlements) is encoded to json with snake case - which turns "Pro" into "pro".

```
(lldb) po customerInfo
<CustomerInfo:
originalApplicationVersion=35,
latestExpirationDate=Optional(2022-06-08 17:58:04 +0000),
activeEntitlements=["Pro": "<EntitlementInfo: \"\nidentifier=Pro,\nisActive=true,\nwillRenew=false,\nperiodType=PeriodType(rawValue: 0),\nlatestPurchaseDate=Optional(2018-02-23 09:40:18 +0000),\noriginalPurchaseDate=Optional(2018-02-23 09:40:18 +0000),\nexpirationDate=nil,\nstore=Store(rawValue: 0),\nproductIdentifier=com.cherpake.macrc.all,\nisSandbox=false,\nunsubscribeDetectedAt=nil,\nbillingIssueDetectedAt=nil,\nownershipType=PurchaseOwnershipType(rawValue: 0)\n>"],
activeSubscriptions=["com.cherpake.control.1msub": "expiresDate: Optional(2022-06-08 17:58:04 +0000)"],
nonSubscriptionTransactions=[<RCStoreTransaction: 0x281384ea0>, <RCStoreTransaction: 0x281385440>, <RCStoreTransaction: 0x2813853b0>],
requestDate=2022-06-02 18:48:28 +0000,
firstSeen=2022-05-08 17:57:30 +0000,
originalAppUserId=$RCAnonymousID:47027854016b41a9a201dcaa0d643615,
entitlements=["Pro": "<EntitlementInfo: \"\nidentifier=Pro,\nisActive=true,\nwillRenew=false,\nperiodType=PeriodType(rawValue: 0),\nlatestPurchaseDate=Optional(2018-02-23 09:40:18 +0000),\noriginalPurchaseDate=Optional(2018-02-23 09:40:18 +0000),\nexpirationDate=nil,\nstore=Store(rawValue: 0),\nproductIdentifier=com.cherpake.macrc.all,\nisSandbox=false,\nunsubscribeDetectedAt=nil,\nbillingIssueDetectedAt=nil,\nownershipType=PurchaseOwnershipType(rawValue: 0)\n>"]
>

(lldb) po String(data: jsonData, encoding: .utf8)
▿ Optional<String>
  - some : "{\"schema_version\":\"3\",\"subscriber\":{\"original_application_version\":\"35\",\"first_seen\":\"2022-05-08T17:57:30Z\",\"subscriptions\":{\"com.cherpake.control.1ysub\":{\"is_sandbox\":false,\"period_type\":\"normal\",\"unsubscribe_detected_at\":\"2022-05-08T17:58:42Z\",\"ownership_type\":\"PURCHASED\",\"expires_date\":\"2017-11-07T12:13:06Z\",\"original_purchase_date\":\"2017-07-04T06:09:53Z\",\"purchase_date\":\"2017-11-05T18:19:43Z\",\"store\":\"app_store\"},\"com.cherpake.control.1msub\":{\"is_sandbox\":false,\"period_type\":\"normal\",\"unsubscribe_detected_at\":\"2022-05-08T17:58:42Z\",\"ownership_type\":\"PURCHASED\",\"expires_date\":\"2022-06-08T17:58:04Z\",\"original_purchase_date\":\"2017-07-04T06:09:53Z\",\"purchase_date\":\"2022-05-08T17:58:04Z\",\"store\":\"app_store\"}},\"management_url\":\"https:\\/\\/apps.apple.com\\/account\\/subscriptions\",\"original_purchase_date\":\"2013-04-13T06:05:24Z\",\"entitlements\":{\"pro\":{\"product_identifier\":\"com.cherpake.macrc.all\",\"purchase_date\":\"2018-02-23T09:40:18Z\"}},\"original_app_user_id\":\"$RCAnonymousID:47027854016b41a9a201dcaa0d643615\",\"non_subscriptions\":{\"com.cherpake.macrc.adsfree\":[{\"id\":\"3e0325e864\",\"purchase_date\":\"2016-04-05T00:19:08Z\",\"store\":\"app_store\",\"original_purchase_date\":\"2016-04-05T00:19:08Z\",\"is_sandbox\":false}],\"com.cherpake.macrc.apps.trial\":[{\"id\":\"6e2f16ddd4\",\"purchase_date\":\"2015-11-25T05:43:20Z\",\"store\":\"app_store\",\"original_purchase_date\":\"2015-11-25T05:43:20Z\",\"is_sandbox\":false}],\"com.cherpake.macrc.all\":[{\"id\":\"0d6bbaed49\",\"purchase_date\":\"2018-02-23T09:40:18Z\",\"store\":\"app_store\",\"original_purchase_date\":\"2018-02-23T09:40:18Z\",\"is_sandbox\":false}]}},\"request_date\":\"2022-06-02T18:48:28Z\"}"
```

CustomerInfoManager:178

**Comment by aboedo** [comment]
Hi! 👋 thanks for reporting!! We're aware of this and working on a bug fix

**Comment by aboedo** [comment]
#1546 is the current workaround

**Comment by cherpake** [comment]
I’m looking at the conversion on my phone and that’s why I probably missing the workaround…

Forgive me but I don’t get it - why the SDK decodes the server response and then stores it as json data, instead of storing the data received from the server in the cache directly…

**Comment by NachoSoto** [comment]
I think @aboedo meant #1651 as the workaround.
But either way the fix is now out in version 4.4.1 and 4.5.1.

**Comment by aboedo** [comment]
Closing this issue as fixed, please check out 4.4.1 or 4.5.1 to get the fix. 

Thanks again @cherpake for reporting!
